### PR TITLE
Flutter Component 0.0.2

### PR DESCRIPTION
**Pull request by rachana-egov**

### CODE REVIEW

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/table_cell.dart [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        return DigitTextFormInput(
+          initialValue: column.value.toString(),
+        );
+      case ColumnType.dropDown:
+        return const DigitDropdown(
+          items: [
+            DropdownItem(name: 'text1', code: '1'),
+            DropdownItem(name: 'text2', code: '2')
+          ],
+        );
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/table_cell.dart]: resolved

File: flutter/digit-ui-components/digit_components/lib/theme/ComponentTheme/bread_crumb_theme.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      separatorTextStyle: TextStyle.lerp(separatorTextStyle, other.separatorTextStyle, t),
+      itemPadding: EdgeInsets.lerp(itemPadding, other.itemPadding, t),
+    );
+
+  }
+  String _lerpString(String a, String b, double t) {
+    /// Interpolate between two strings `a` and `b` based on `t`
+    /// For simplicity, you can directly return one of the strings if `t` is 0 or 1
+    return t < 0.5 ? a : b;
+  }
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/theme/ComponentTheme/bread_crumb_theme.dart:69]: done

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_search_bar.dart [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                focusedBorder: InputBorder.none,
+                enabledBorder: InputBorder.none,
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_search_bar.dart:89]: did that to make it more customizable

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_tree_select_dropdown.dart [hunk 4]
```diff
@@ -1,60 +1,46 @@
...
+///      selectedOptions = selectedItems;
+///    });
+///  },
+///  // Customize the appearance of chips
+///  chipConfig: ChipConfig(),
+///  // Customize the suffix icon (dropdown arrow)
+///  suffixIcon: Icon(Icons.arrow_drop_down),
+///  // Customize the selection type (multiSelect or singleSelect)
+///  treeSelectionType: TreeSelectionType.multiSelect,
+///),
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_tree_select_dropdown.dart:25]: done

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_dob_picker.dart [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+  // Calculate the date from the given age in years and months
+  DateTime _getDateFromAge() {
+    int years = int.tryParse(yearController.text) ?? 0;
+    int months = int.tryParse(monthController.text) ?? 0;
+    DateTime now = DateTime.now();
+
+    DateTime dob = DateTime(now.year - years, now.month - months);
+    return dob;
+  }
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_dob_picker.dart:95]: age cann't be more than current date

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_otp.dart [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+    _pin = List.generate(widget.length, (int i) => '');
+  }
+
+  @override
+  void dispose() {
+    for (var controller in _textControllers) {
+      controller?.dispose();
+    }
+    super.dispose();
+  }
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_otp.dart:126]: done

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_base_form_input.dart [hunk 7]
```diff
@@ -213,28 +232,45 @@ class BaseDigitFormInputState extends State<BaseDigitFormInput> {
...
   }
 
   @override
   void dispose() {
     super.dispose();
+    if (widget.controller == null) {
+      _controller.dispose();
+    }
     myFocusNode.removeListener(onFocusChange);
     myFocusNode.dispose();
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_base_form_input.dart:268]: done

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/upload_image.dart [hunk 8]
```diff
@@ -0,0 +1,493 @@
...
+              onCrossTap: () {
+                Navigator.of(context).pop();
+              },
+              title: widget.cameraTitle ?? 'Camera',
+              type: PopUpType.simple,
+              actions: [
+                DigitButton(
+                  mainAxisSize: MainAxisSize.max,
+                  prefixIcon: Icons.camera_enhance,
+                  label: widget.captureText ??'Capture',
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/widgets/atoms/upload_image.dart:78]: use i18 code

File: flutter/digit-ui-components/digit_components/lib/widgets/helper_widget/overlay_dropdown.dart [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+                    color: const DigitColors().light.paperPrimary,
+                    child: Container(
+                      width: dropdownWidth,
+                      decoration: BoxDecoration(
+                        boxShadow: [
+                          BoxShadow(
+                            offset: Offset(0, widget.type == OverlayDropdownType.header ? -1 : 1),
+                            blurRadius: 4.4,
+                            spreadRadius: 0,
+                            color: const Color(0x26000000), // #00000026
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/widgets/helper_widget/overlay_dropdown.dart:115]: Color should be read from Themes

File: flutter/digit-ui-components/digit_components/lib/widgets/molecules/digit_timeline_molecule.dart [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+  final String hideDetailText;
+  final bool capitalizedLetter;
+  final int initialStepCount;
+  final bool showAllSteps; // New prop to show all steps initially
+
+  const TimelineMolecule({
+    Key? key,
+    required this.steps,
+    this.viewDetailText = 'View Details',
+    this.hideDetailText = 'Hide Details',
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/widgets/molecules/digit_timeline_molecule.dart:18]: Use i18 codes

File: flutter/digit-ui-components/digit_components/lib/widgets/molecules/language_selection_card.dart [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+    final TextPainter textPainter = TextPainter(
+      textDirection: TextDirection.ltr,
+    );
+
+    double maxWidth = 0;
+    for (var item in widget.rowItems) {
+      textPainter.text = TextSpan(
+        text: item.label,
+        style: const TextStyle(
+          fontSize: 16,
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/widgets/molecules/language_selection_card.dart:114]: Font size should be read from Themes

**Review by naveen-egov — CHANGES_REQUESTED** [review]
Please make the necessary changes as requested

File: flutter/digit-ui-components/digit_components/CHANGELOG.md [hunk 12]
```diff
@@ -5,3 +5,33 @@
 ## 0.0.2-beta
 
 * test release.
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/CHANGELOG.md:7]: updated with previous release,

File: flutter/digit-ui-components/digit_components/lib/blocs/TimeSelectionBloc.dart [hunk 13]
```diff
@@ -5,11 +5,15 @@ class TimeSelectionBloc {
...
   }) async {
     /// Show a time picker and update the controller's value
     TimeOfDay? selectedTime = await showTimePicker(
       context: context,
       initialTime: TimeOfDay.now(),
+      confirmText: confirmText,
+      cancelText: cancelText,
       builder: (BuildContext context, Widget? child) {
         return MediaQuery(
           data: MediaQuery.of(context).copyWith(alwaysUse24HourFormat: false),
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/blocs/TimeSelectionBloc.dart]: it is from params only

File: flutter/digit-ui-components/digit_components/lib/constants/app_constants.dart [hunk 14]
```diff
@@ -1,20 +1,33 @@
+import 'package:digit_ui_components/digit_components.dart';
 import 'package:flutter/material.dart';
+import '../theme/theme.dart';
 
-import '../theme/colors.dart';
-import '../theme/digit_theme.dart';
-
-class Default{
+class Base{
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/constants/app_constants.dart:5]: done

File: flutter/digit-ui-components/digit_components/lib/theme/ComponentTheme/action_card_theme.dart [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+            color: const Color(0xFF000000).withOpacity(.16),
+            offset: const Offset(0, 1),
+            spreadRadius: 0,
+            blurRadius: 2,
+          ),
+        ],
+      ),
+      padding: EdgeInsets.all(theme.spacerTheme.spacer6),
+      width: isMobile ? 328.0 : 548.0,
+      spacing: 24,
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/theme/ComponentTheme/action_card_theme.dart:44]: done

File: flutter/digit-ui-components/digit_components/lib/utils/fileService/web_file_download.dart [hunk 16]
```diff
@@ -0,0 +1,12 @@
+import 'dart:typed_data';
+import 'package:universal_html/html.dart' as html;
+import 'package:digit_ui_components/widgets/atoms/upload_drag.dart';
+
+class WebFileDownloader {
+  Future<void> downloadFile(DroppedFile file) async {
+    html.AnchorElement(
+        href: html.Url.createObjectUrlFromBlob(html.Blob([file.data])))
+      ..setAttribute("download", file.name)
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/utils/fileService/web_file_download.dart]: that is html default down attribute

File: flutter/digit-ui-components/digit_components/lib/utils/validators/file_validator.dart [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+
+String? validateFile(PlatformFile file, List<FileValidator> validators) {
+  // Perform validation for the given file using the provided validators
+  for (var validator in validators) {
+    switch (validator.type) {
+      case FileValidatorType.fileType:
+        if (!_isFileTypeAllowed(file, validator.value)) {
+          return validator.errorMessage ?? 'Invalid file type';
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/utils/validators/file_validator.dart:24]: they case send a error message to localize and this is not a stateful class just a function

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_back_button.dart [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+
+class BackNavigationButton extends StatelessWidget {
+  final VoidCallback? handleBack;
+  final String backButtonText;
+  final BackNavigationButtonThemeData? backNavigationButtonThemeData;
+
+  const BackNavigationButton({
+    super.key,
+    this.handleBack,
+    this.backButtonText = 'Back',
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_back_button.dart]: done

File: flutter/digit-ui-components/digit_components/lib/widgets/molecules/language_selection_card.dart (hunk 11 above)
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/molecules/language_selection_card.dart:114]: done

File: flutter/digit-ui-components/digit_components/lib/widgets/molecules/digit_timeline_molecule.dart (hunk 10 above)
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/molecules/digit_timeline_molecule.dart:18]: done

**Review by naveen-egov — APPROVED** [review]

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_tag.dart [hunk 19]
```diff
@@ -0,0 +1,145 @@
...
+              : null,
+        ),
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            if (widget.isIcon && !(widget.type == TagType.monochrome && widget.customIcon == null)) ...[
+              widget.customIcon ?? icon!,
+              SizedBox(width: theme.spacerTheme.spacer3),
+            ],
```
[the remark this answers is not in the document]
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_tag.dart:120]: i have made sure that icon is not null in first check

**Review by naveen-egov — APPROVED** [review]

**Review by naveen-egov — APPROVED** [review]

File: react/ui-components/src/hoc/stories/DropDown.stories.js
[the remark this answers is not in the document]
Comment (rachana-egov) [reply react/ui-components/src/hoc/stories/DropDown.stories.js:1]: resolved

**Pull request merged**